org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/registry/MavenProjectManager.java

```diff
@@ -47,7 +47,7 @@
  * dependency between ProjectRegistryRefreshJob and ProjectRegistryManager. Otherwise, all requests are forwarded to
  * ProjectRegistryManager as is.
  */
-@Component(service = {IMavenProjectRegistry.class})
+@Component(service = {MavenProjectManager.class, IMavenProjectRegistry.class})
```

Comment:
Is MavenProjectManager used anywhere as a service? Actually we should avoid registering implementations as services and instead use e.g. Whiteboardpattern instead.

Reply:
Is is used by WorkspaceStateWriter so that's why I have to add it there

Reply:
Then probably this has to be reversed, `WorkspaceStateWriter` should not use the `MavenProjectManager` but `MavenProjectManager` should collect all `IMavenProjectChangedListener`

Reply:
Then I don't see how `MavenProjectManager` is ever needed and we probobly better find an alternative way to get workspace state file location (even duplicating some code or using a static method to build its name).